scheds/rust/scx_layered/src/bpf/main.bpf.c

```diff
@@ -1079,7 +1098,14 @@ s32 BPF_STRUCT_OPS(layered_select_cpu, struct task_struct *p, s32 prev_cpu, u64
 	if (taskc->layer_id == MAX_LAYERS || !(layer = lookup_layer(taskc->layer_id)))
 		return prev_cpu;
 
-	cpu = pick_idle_cpu(p, prev_cpu, cpuc, taskc, layer, true);
+	if (layer->periodically_refresh && taskc->layer_refresh_seq < layer_refresh_seq_avgruntime)
+		taskc->refresh_layer = true;
```

Comment:
It probably would be better to do this in `pick_idle_cpu()` as select_cpu() may be skipped in some cases.